### DISCUSSION TITLE
Can Facebook or Maskbook add me as a friend and then decrypt my posts?

### DISCUSSION THREAD

**Discussion by trymeouteh** [DimensionDev/Maskbook#2120] [General]
Do you need to be friends on facebook in order to share keys to decrypt each others posts? Or is there a "friends list" on the mask network which is used to decrypt. I wonder if it is possible for facebook to add a friend who uses maskbook and then can read my encrypted posts or if maskbook can add/remove a friend to decrypt posts

**Comment by Tedko** [comment]
If you chose ‘Everyone’ it’s an AES key. So whoever have the AES key can
decrypt. But for select specific recipients it’s not. Try it out?

**Comment by Tedko** [comment]
I’ll go sleep now but I believe our engineers will answer u tmr. Thx

**Reply by Jack-Works** [reply]
I cannot understand what he exactly wants to ask about

**Reply by trymeouteh** [reply]
How do I ensure only my facebook friends get the key to decrypt my posts and no one else, including facebook and maskbook?

**Reply by Jack-Works** [reply]
We're using end-to-end encryption. The only friend you choose will receive the encrypted post key.

**Reply by trymeouteh** [reply]
> We're using end-to-end encryption. The only friend you choose will receive the encrypted post key.

Is this encrypted post key shared automatically be detecting who my facebook friends are and seeing if they have maskbook and therefore sharing my post key? Or do I have to manually share the post key to them in an email or something like a PGP public key?

**Reply by yisiliu** [reply]
![image](https://user-images.githubusercontent.com/3343358/102678123-54bc3680-41e1-11eb-8385-757314c70101.png)
You can refer to this diagram to understand this PGP-like encryption process. All you need to do is to enclose your public key in your 
 Facebook or Twitter account bio or in an encrypted post, where your public key will be automatically added in the payload FYI.
